internal/controller/pool_controller.go

```diff
@@ -336,16 +329,10 @@ func (r *PoolReconciler) ensureFinalizer(ctx context.Context, pool *garmoperator
 	return nil
 }
 
-func (r *PoolReconciler) comparePoolSpecs(ctx context.Context, pool *garmoperatorv1alpha1.Pool, poolClient garmClient.PoolClient) (bool, []params.Instance, error) {
+func (r *PoolReconciler) comparePoolSpecs(ctx context.Context, pool *garmoperatorv1alpha1.Pool, image *garmoperatorv1alpha1.Image, poolClient garmClient.PoolClient) (bool, []params.Instance, error) {
```

Comment:
as we only use `image.Spec.Tag` (string) we could also get rid of the entire image CR object and just pass the image-tag string as argument in here